lib/Flags.js

```diff
@@ -0,0 +1,284 @@
+// @flow
+
+const Report = require('./Report.js');
+
+function help() /*: string */ {
+  return `
+elm-test init
+    Create example tests
+
+elm-test
+    Run tests in the tests/ folder
+
+elm-test TESTFILES
+    Run TESTFILES, for example "src/**/*Tests.elm"
+
+elm-test install PACKAGE
+    Like \`elm install PACKAGE\`, except it installs to
+    "test-dependencies" in your elm.json
+
+Options:
+    --compiler /path/to/compiler
+        Use a custom path to an Elm executable. Default: elm
+    --seed INT
+        Run with a previous fuzzer seed. Default: A random seed
+    --fuzz INT
+        Run with each fuzz test performing this many iterations. Default: 100
+    --report json
+    --report junit
+    --report console
+        Print results to stdout in the given format. Default: console
+    --version
+        Print version and exit
+    --watch
+        Run tests on file changes
+  `.trim();
+}
+
+const longOptionWithValue = /^(--[^-=][^=]*)=([^]*)$/;
+const looksLikeOption = /^--?[^-]/;
+
+// Poor man’s type alias. We can’t use /*:: type Command = ... */ because of:
+// https://github.com/prettier/prettier/issues/2597
+const Command /*:
+  | { tag: 'help' }
+  | { tag: 'version' }
+  | { tag: 'init' }
+  | { tag: 'install', packageName: string }
+  | { tag: 'make', testFileGlobs: Array<string> }
+  | { tag: 'test', testFileGlobs: Array<string> } */ = { tag: 'help' };
+void Command;
+
+function parseArgv(
+  argv /*: Array<string> */
+) /*: Result<
+  string,
+  {
+    command: typeof Command,
+    options: {
+      compiler: string | void,
+      fuzz: number,
+      report: typeof Report.Report,
+      seed: number,
+      watch: boolean,
+    },
+  }
+> */ {
+  const options = {
+    help: false,
+    version: false,
+    watch: false,
+    compiler: undefined,
+    seed: Math.floor(Math.random() * 407199254740991) + 1000,
+    fuzz: 100,
+    report: 'console',
+  };
+
+  const rest = [];
+  let raw = [];
+
+  for (let index = 0; index < argv.length; index++) {
+    const fullArg = argv[index];
+    const match = longOptionWithValue.exec(fullArg);
+    const arg = match === null ? fullArg : match[1];
+
+    // Get the value passed to a long flag, or `fallback` if there is none.
+    // This is either the part after the equal sign: `--flag=value`.
+    // Or the next argument: `--flag value`.
+    const getValue = (fallback /*: string */) /*: string */ => {
+      if (match === null) {
+        index++;
```

Comment:
I do not like that we have a for loop and then a closure that mutates the index variable. My lunch break is almost over though and I haven't come up with a more constructive suggestion yet. Let me go and think about this.

Reply:
We talked about this on discord. @lydell is going to look into some alternatives to minimist